dicesGame/src/main/java/com/example/dices/services/DiceService.java

```
@@ -0,0 +1,88 @@
+package com.example.dices.services;
+
+import com.example.dices.models.DiceModel;
+import com.example.dices.repositories.IDiceRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+
+@Service
+public class DiceService {
+    private final IDiceRepository diceRepository;
+
+    @Autowired
+    public DiceService(IDiceRepository diceRepository) {
+        this.diceRepository = diceRepository;
+    }
+
+    public static class ServiceConstants {
+        public static final int MINIMUM_DICE_SIZE = 1;
+        public static final int MAXIMUM_DICE_SIZE = 1000;
+    }
+
+    // show list of created dices
+    public List<DiceModel> getDices() {
+        return (List<DiceModel>) diceRepository.findAll();
+    }
+
+
+    // create dice in the database
+    public DiceModel createDice(DiceModel dice) {
+        int validSize = dice.getDiceSize();
+        if (validSize >= ServiceConstants.MINIMUM_DICE_SIZE &&
+                validSize <= ServiceConstants.MAXIMUM_DICE_SIZE) {
+            diceRepository.save(dice);
+            return dice;
+        } else {
+            return null;
```

Please throw an exception instead of returning `null`.
We should avoid using `null` as much as possible.

---

"null" was not used in the "getById" service. The Controller and unit tests are modified.
The other services are pending modification.

---

nowhere else in the code is "null" used